### PR TITLE
Add multi button support.

### PR DESCRIPTION
**Pull request by Luitwieler**
Updated documentation, added the logic for multi button support and included wildcards for bundleIndentifier. 

A pull request for the the following topic: https://github.com/lukasroegner/homebridge-apple-tv-remote/issues/62#issuecomment-699543137

### CODE REVIEW

**Comment by Luitwieler** [comment]
@lukasroegner I created a PR however I see that we have some conflicts.

**Comment by Luitwieler** [comment]
@lukasroegner Conflicts are now resolved and includes the changes for stabilizing the playing state while keeping the logic for current app active.

**Comment by lukasroegner** [comment]
Thanks, I'll review the changes!

**Comment by Luitwieler** [comment]
Thanks! Let me know if you have any questions. Never programmed TypeScript before, but I tested the code on tvOS 14.2 and seems to be working fine.

File: src/lib/controllers/apple-tv-controller.ts [hunk 1]
```diff
@@ -63,29 +63,39 @@ export class AppleTvController {
...
-            const onCharacteristic = playPauseSwitchService.useCharacteristic<boolean>(Homebridge.Characteristics.On);
-            onCharacteristic.valueChanged = newValue => {
-                if (onCharacteristic.value !== newValue) {
-                    platform.logger.info(`[${deviceConfiguration.name}] Play/pause switch changed to ${newValue}`);
-                    try {
-                        if (newValue) {
-                            client.play();
-                        } else {
-                            client.pause();
+            deviceConfiguration.appPlayPauseSwitches.forEach(PlayPauseSwitch => {
```
Comment (lukasroegner) [inline src/lib/controllers/apple-tv-controller.ts:66]: - null/undefined check is missing. I would suggest the same pattern as with the command switches.
- Style: use for-of-loop

```
if (deviceConfiguration.appPlayPauseSwitches && deviceConfiguration.appPlayPauseSwitches.length > 0) {
    for (let appPlayPauseSwitchConfiguration of deviceConfiguration.appPlayPauseSwitches) {
        ...
    }
}
```

File: src/lib/controllers/apple-tv-controller.ts [hunk 2]
```diff
@@ -63,29 +63,39 @@ export class AppleTvController {
...
-            // Subscribes for events of the client
-            client.on('isPlayingChanged', _ => {
-                platform.logger.debug(`[${deviceConfiguration.name}] Play/pause switch updated to ${client.isPlaying}`);
-                onCharacteristic.value = client.isPlaying;
+                };
+
+                // Subscribes for events of the client
+                client.on('isPlayingChanged', _ => {
+                    platform.logger.debug(`[${deviceConfiguration.name}] Play/pause switch updated to ${client.isPlaying}`);
+                    if(PlayPauseSwitch.bundleIdentifier == client.currentApp || PlayPauseSwitch.bundleIdentifier == "*")
```
Comment (lukasroegner) [inline src/lib/controllers/apple-tv-controller.ts:89]: Style:
```
if(PlayPauseSwitch.bundleIdentifier == client.currentApp || PlayPauseSwitch.bundleIdentifier == "*") {
  onCharacteristic.value = client.isPlaying;
} else {
  onCharacteristic.value = false;
}
```

**Review by lukasroegner — COMMENTED** [review]
Thank you so much for the PR, this is a clean and simple solution! I've added some comments, mostly minor things (comments, styles, etc.).

One thing that is missing:
The `platform.ts`, line 71 and 76, is missing checks if the array of app play/pause switches is provided.

**Comment by Luitwieler** [comment]
I pushed the changes to the branch and should resolve all the comments. Thanks for the feedback!

File: README.md [hunk 3]
```diff
@@ -73,6 +73,13 @@ npm install -g homebridge-apple-tv-remote
                     "onOffTvName": "<CUSTOM-NAME>",
                     "isPlayPauseSwitchEnabled": false,
                     "playPauseSwitchName": "<CUSTOM-NAME>",
+                    "appPlayPauseSwitches": [
+                        {
+                            "name": "<BUTTON-NAME>",
+                            "bundleIdentifier": "<APP-IDENTIFIER>" 
+                        },
+
```
Comment (lukasroegner) [inline README.md:81]: Style: remove empty line

File: src/lib/configuration/appplaypause-configuration.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Represents an app specific Play/Pause switch in the homebridge configuration for the plugin.
+ */
+export interface AppPlayPauseSwitchConfiguration {
+
+    name: string;
```
Comment (lukasroegner) [inline src/lib/configuration/appplaypause-configuration.ts:6]: Style: comment missing. Suggestion:

```
/**
 * Gets or sets a unique name for the switch.
 */
```

File: src/lib/configuration/appplaypause-configuration.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+
+    bundleIdentifier: string;
```
Comment (lukasroegner) [inline src/lib/configuration/appplaypause-configuration.ts:8]: Style: comment missing. Suggestion:

```
/**
 * Gets or sets the bundle identifier of the app that this switch represents, e.g. 'com.netflix.Netflix' or 'com.google.ios.youtube'.
 */
```

File: src/lib/configuration/appplaypause-configuration.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+}
```
Comment (lukasroegner) [inline src/lib/configuration/appplaypause-configuration.ts:9]: Style: empty line at the end of each file required

File: src/lib/configuration/appplaypause-configuration.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+/**
```
Comment (lukasroegner) [inline src/lib/configuration/appplaypause-configuration.ts:1]: Style: filename should be `app-play-pause-switch-configuration.ts` (kebab-case of interface name).

File: src/lib/controllers/apple-tv-controller.ts [hunk 8]
```diff
@@ -60,33 +60,71 @@ export class AppleTvController {
             });
         }
 
-        // Creates the Play/Pause switch if requested
-        if (deviceConfiguration.isPlayPauseSwitchEnabled) {
+                // Creates the Play/Pause switch if requested
```
Comment (lukasroegner) [inline src/lib/controllers/apple-tv-controller.ts:63]: Style: wrong indent

File: src/lib/controllers/apple-tv-controller.ts [hunk 9]
```diff
@@ -60,33 +60,71 @@ export class AppleTvController {
...
-            const onCharacteristic = playPauseSwitchService.useCharacteristic<boolean>(Homebridge.Characteristics.On);
-            onCharacteristic.valueChanged = newValue => {
-                if (onCharacteristic.value !== newValue) {
-                    platform.logger.info(`[${deviceConfiguration.name}] Play/pause switch changed to ${newValue}`);
-                    try {
-                        if (newValue) {
-                            client.play();
-                        } else {
-                            client.pause();
+
```
Comment (lukasroegner) [inline src/lib/controllers/apple-tv-controller.ts:96]: Style: unnecessary empty lines

File: src/lib/controllers/apple-tv-controller.ts [hunk 10]
```diff
@@ -60,33 +60,71 @@ export class AppleTvController {
...
-                    } catch (e) {
-                        platform.logger.warn(`[${deviceConfiguration.name}] Failed to change play/pause to ${newValue}`);
                     }
-                }
+                };
+
+                // Subscribes for events of the client
+                client.on('isPlayingChanged', _ => {
+                    platform.logger.debug(`[${deviceConfiguration.name}] Play/pause switch updated to ${client.isPlaying}`);
+                    if(appPlayPauseSwitchConfiguration.bundleIdentifier == client.currentApp){
```
Comment (lukasroegner) [inline src/lib/controllers/apple-tv-controller.ts:120]: - Style: space between `if` and `(` missing
- Style: space between `)` and `{` missing

File: src/lib/controllers/apple-tv-controller.ts [hunk 11]
```diff
@@ -60,33 +60,71 @@ export class AppleTvController {
...
+                        onCharacteristic.value = client.isPlaying;
+                    } else{
```
Comment (lukasroegner) [inline src/lib/controllers/apple-tv-controller.ts:122]: Style: space between `else` and `{` missing

**Review by lukasroegner — COMMENTED** [review]
There are some style issues, otherwise perfect! I'll merge to master after those minor things are resolved. Thanks again!

**Comment by Luitwieler** [comment]
@lukasroegner Done, ready to be merged to Master!

**Pull request merged**